pkg/kapp/clusterapply/delete_change.go

```diff
@@ -110,3 +111,11 @@ func (c DeleteOrphanStrategy) Apply() error {
 	_, err = c.d.identifiedResources.Patch(c.res, types.JSONPatchType, patchJSON)
 	return err
 }
+
+func (c DeleteChange) buildDescMsg(res ctlres.Resource, doneApplying bool) []string {
```

Comment:
in addition to adding code here, take a look at https://github.com/vmware-tanzu/carvel-kapp/blob/develop/pkg/kapp/resourcesmisc/deleting.go as well. it needs to also contain similar code. and once they are both have same code... can we instantiate Deleting object here to reuse it.

Reply:
Done

Reply:
Doing this way what I observed is at this point in time resource doesn't get marked as deleted and hence `resource.IsDeleting()`  returns false and that's why `NewDeleting(res)`  method returns nil.

Problem:
Deploying a resource with finalizer and trying to delete it first time this message `waiting on finalizers ...`  doesn't come up as when doing kapp delete  first time resource doesn't get marked as deleted and hence `NewDeleting(res)`  method returns `nil` and hence it doesn't call deleting method to print the waiting message.  (while on doing kapp delete second time waiting on finalizers message shows up)